File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/2506. Count Pairs Of Similar Strings/Python soultion/2506. Count Pairs Of Similar Strings.py
class Solution:
    def similarPairs(self, words: List[str]) -> int:
        count = 0
        # brute force solution        
        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                if set(words[i]) == set(words[j]):
                    count += 1
        return count

File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/2506. Count Pairs Of Similar Strings/C++ solution/2506. Count Pairs Of Similar Strings.cpp
class Solution {
public:
    int similarPairs(vector<string>& words) {
        int count = 0;
        // brute force solution
        for (int i = 0; i < words.size(); i++) {
            for (int j = i + 1; j < words.size(); j++) {
                set<char> set1(words[i].begin(), words[i].end());
                set<char> set2(words[j].begin(), words[j].end());
                if (set1 == set2) {
                    count += 1;
                }
            }
        }
        return count;
    }
};